components/provider_cldr/src/transform/dates.rs

```diff
@@ -115,6 +115,83 @@ impl From<&cldr_json::StylePatterns> for gregory::patterns::StylePatternsV1 {
     }
 }
 
+impl From<&cldr_json::DateTimeFormats> for gregory::patterns::DateTimeFormatsV1 {
+    fn from(other: &cldr_json::DateTimeFormats) -> Self {
+        use gregory::patterns::{
+            PatternV1, SkeletonFieldsV1, SkeletonFieldsV1Error, SkeletonTupleV1,
+        };
+
+        // TODO(#308): Support numbering system variations. We currently throw them away.
+        Self {
+            style_patterns: gregory::patterns::StylePatternsV1 {
+                full: other.full.get_pattern().clone(),
+                long: other.long.get_pattern().clone(),
+                medium: other.medium.get_pattern().clone(),
+                short: other.short.get_pattern().clone(),
+            },
+            skeletons: {
+                let mut skeleton_tuples: Vec<SkeletonTupleV1> = Vec::new();
+
+                // The CLDR keys for available_formats can have duplicate skeletons with either
+                // an additional variant, or with multiple variants for different plurals.
+                for (skeleton_str, pattern_str) in other.available_formats.0.iter() {
+                    let mut unique_skeleton = String::new();
+                    let mut variant_text = String::new();
+
+                    // Split out the unique skeleton component, and the variant text.
+                    //   "MMMMW-count-two"
+                    //   unique_skeleton: "MMMMW"
+                    //   variant_text: "-count-two"
+                    let chars = skeleton_str.chars();
+                    let mut target = &mut unique_skeleton;
+                    for ch in chars {
+                        if ch == '-' {
+                            target = &mut variant_text;
+                        }
+                        target.push(ch);
+                    }
+
+                    let skeleton_fields_v1 =
+                        match SkeletonFieldsV1::try_from(unique_skeleton.as_str()) {
+                            Ok(s) => s,
+                            Err(err) => match err {
+                                // Ignore unimplemented fields for now.
+                                SkeletonFieldsV1Error::SymbolUnimplemented(_) => continue,
+                                SkeletonFieldsV1Error::SymbolUnknown(symbol) => panic!(
```

Comment:
I have not yet formed a strong opinion on whether we should `panic!` in transformer code.  I've usually been returning errors from functions instead of panicking, but I realize that maybe it's acceptable to panic in transformer code, because it helps better identify where the error originated.  But I do think we should be consistent.  @Manishearth?

Reply:
Follow-up: #574